clustermesh-apiserver/health/health.go

```diff
@@ -71,3 +76,31 @@ func registerHealthAPIServer(lc cell.Lifecycle, params parameters) {
 		OnStop: func(ctx cell.HookContext) error { return srv.Shutdown(ctx) },
 	})
 }
+
+type healthParameters struct {
+	cell.In
+
+	SyncState syncstate.SyncState
+	Logger    logrus.FieldLogger
+}
+
+func healthEndpoints(params healthParameters) []EndpointFunc {
```

Comment:
I'm personally a bit on the fence on the opportunity of merging this function, instead of keeping it separate for each component. The rationale for having each component defining its own `healthEndpoints` is that the checks may differ (e.g., the clustermesh-apiserver may additionally check connectivity to Kubernetes, which does not make sense for KVStoreMesh). Hence, having separate functions would make extending the checks easier. That said, I also reckon that the two functions are identical at the moment.

Reply:
It logically makes sense for them to be different, so I split this such that the API server itself is common but the health endpoint declarations like this remain split. That should help with any future changes to extend these APIs.